### PR TITLE
[composite compliance] skip stride check if dim size is 0 or 1

### PR DESCRIPTION
**Pull request by kshitij12345**
`at::for_blob` short-circuits if first dim has size `0` or if number of dims is greater than `1`.
`at::for_blob` is used in `_make_wrapper_subclass` implementation and results in `SubclassTensor` having different stride for the above case.

So we skip stride consistency check based on the above condition.

<details> 

```python
import torch
from torch.testing._internal.composite_compliance import generate_cct
elem = torch.ones(0,).as_strided((0,), (2,), 0)
CCT = generate_cct()
ret = CCT(elem)
print(f"{ret.shape = }, {ret.stride() = }, {ret.elem.shape = }, {ret.elem.stride() = }")
```

Output
```
ret.shape = torch.Size([0]), ret.stride() = (1,), ret.elem.shape = torch.Size([0]), ret.elem.stride() = (2,)
```

</details>

### CODE REVIEW

File: torch/testing/_internal/composite_compliance.py [hunk 1]
```diff
@@ -44,6 +44,11 @@ def check_metadata_consistency(wrapper_tensor, CCT):
         'storage_offset': Tensor.storage_offset,
     }
     for metadata_name, metadata_accessor in things_to_check.items():
+        # stride when a dimension size is `0` or `1` dont matter,
+        # and few functions take advantange of this to short-circuit
+        if metadata_name == 'stride' and \
+                (0 == wrapper_tensor.shape[0] or len(wrapper_tensor.shape) == 1):
+            continue
```
Comment (kshitij12345) [inline torch/testing/_internal/composite_compliance.py:51]: Condition based on https://github.com/pytorch/pytorch/blob/d136852bda4d9eb60927b7c8ce0f665686524643/aten/src/ATen/templates/Functions.cpp#L44-L55
